### ISSUE TITLE
Proc number while executing make

### ISSUE THREAD

**Issue by daveszymczak** [million12/docker-nginx-hhvm#1]
It would be nice to add threads to make command with -j flag, I think core or core + 1 would be quite ok. Something like this

make -j$(($(nproc)+1))

**Comment by ryzy** [comment]
Yeah, that's a valid point... but I was running out of memory on 4GB machine while compiling with such a setting. Therefore, to be on the safe side, I left just `make`. Any ideas how to make it more bullet-proof?

**Comment by ryzy** [comment]
O wow, Dawid, it's you!

**Comment by daveszymczak** [comment]
Yeah, it's me, same old :-) How many cores did you have on your VM?

**Comment by ryzy** [comment]
8 (usually ;-)

**Comment by daveszymczak** [comment]
Was it AWS?

**Comment by ryzy** [comment]
No, local Vagrant machine with CoreOS with NO swap.

I think it was the same on CircleCI where this image is being build (and then pushed to Docker Hub) - they have 36 cores by default and it's on AWS (not sure about swap here)

The thing is it was random: sometimes it went through, sometimes not.

**Comment by ryzy** [comment]
Made a change in https://github.com/million12/docker-nginx-hhvm/tree/nproc
Let's see how it goes on CircleCI: https://circleci.com/gh/million12/docker-nginx-hhvm/14

**Comment by daveszymczak** [comment]
Interesting, let's see then.

Personally I would also try a little bit different approach, in my projects I separate FPM/HHVM from nginx,  so I can which service is handled by what interpreter. Would be same thing here but in completed stack I would stick some load balancer between nginx and PHP interpreter. So in other words, HHVM needs at least 2 cores to compile, so instead of running one with maximum threads I would run x with 2 threads and balance it with some haProxy or something.
